tests/code/test_spreadsheet.py

```diff
@@ -0,0 +1,194 @@
+
+@pytest.mark.integration
+def test_termParticipation():
+    with mainDB.atomic() as transaction:    
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts'], 'CELTS-Sponsored Event': [None]}
+
+        EventParticipant.create(user = 'partont',
+                                event = 10,
+                                hoursEarned = 1)
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts', 'partont'], 'CELTS-Sponsored Event': [None]}
+
+        EventParticipant.create(user = 'ayisie',
+                                event = 14,
+                                hoursEarned = 1)
+        assert termParticipation('Fall 2020')=={'Adopt-a-Grandparent': ['khatts', 'partont'], 'CELTS-Sponsored Event': ['ayisie']}
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_getRetentionRate():
+    with mainDB.atomic() as transaction:
+        assert getRetentionRate("2020-2021") == [('Adopt-a-Grandparent', '0.0%'), ('CELTS-Sponsored Event', '0.0%')]
+        
+        testEvent = Event.create(name="Test Event",
+                                 term=2,
+                                 program=3)
+        EventParticipant.create(user = 'khatts',
+                                event = testEvent,
+                                hoursEarned = 1)
+        assert getRetentionRate("2020-2021") == [('Adopt-a-Grandparent', '100.0%'), ('CELTS-Sponsored Event', '0.0%')]
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_repeatVolunteers():
+    with mainDB.atomic() as transaction:
+
+        assert list(repeatVolunteers().execute()) == [('Sreynit Khatt', 4), ('Zach Neill', 2)]
+        EventParticipant.create(user = 'partont',
+                                event = 2,
+                                hoursEarned = 1)
+        assert list(repeatVolunteers().execute()) == [('Sreynit Khatt', 4), ('Zach Neill', 2), ('Tyler Parton', 2)]
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_repeatVolunteersPerProgram():
+    with mainDB.atomic() as transaction:
+        assert list(repeatVolunteersPerProgram().execute()) == [('Zach Neill', 'Hunger Initiatives', 2), ('Sreynit Khatt', 'Adopt-a-Grandparent', 3)]
+        EventParticipant.create(user = 'partont',
+                                event = 2,
+                                hoursEarned = 1)
+        assert list(repeatVolunteersPerProgram().execute()) == [('Zach Neill', 'Hunger Initiatives', 2), ('Tyler Parton', 'Hunger Initiatives', 2),
+                                                                ('Sreynit Khatt', 'Adopt-a-Grandparent', 3)]
+
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_volunteerMajorAndClass():
+    with mainDB.atomic() as transaction:
+        assert list(volunteerMajorAndClass(User.major).execute()) == [('Biology', 1), ('Chemistry', 1), ('Computer Science', 2), ('Psychology', 1)]
```

Comment:
Can we run each of the assertions before and after the User is created to make sure each one works as expected?